std/range/package.d

```diff
@@ -6678,12 +6677,18 @@ private:
 Given a range of ranges, returns a range of ranges where the $(I i)'th subrange
 contains the $(I i)'th elements of the original subranges.
  */
-Transposed!RangeOfRanges transposed(RangeOfRanges)(RangeOfRanges rr)
+auto ref transposed(RangeOfRanges)(RangeOfRanges rr)
 if (isForwardRange!RangeOfRanges &&
-    isInputRange!(ElementType!RangeOfRanges) &&
-    hasAssignableElements!RangeOfRanges)
+    isInputRange!(ElementType!RangeOfRanges))
 {
-    return Transposed!RangeOfRanges(rr);
+    static if (hasAssignableElements!RangeOfRanges)
+    {
+        return Transposed!RangeOfRanges(rr);
+    }
+    else
+    {
+        return Transposed!(typeof(rr.array))(rr.array);
```

Comment:
We can't accept this. It eagerly copies the range, which is inefficiently and unexpected - entirely out of character for std.range. If a user wants this, they can use `r.array.transposed` themselves.